match2/commons/display_util.lua

```diff
@@ -8,17 +8,29 @@ local DisplayUtil = {propTypes = {}, types = {}}
 Checks that the props to be used by a display component satisfy type
 constraints. Throws if it does not.
 
-For performance reasons, this only checks types of properties, and does not
-check contents inside table properties. Specify options.maxDepth to override
-this behavior.
+For performance reasons, type checking is disabled unless the forceTypeCheck
+developer flag is enabled via {{#vardefine:forceTypeCheck|1}}. Certain other
+developer flags like matchGroupDev can also enable type checking.
+
+By default this only checks types of properties, and not their contents. If
+forceTypeCheck is enabled, then the contents of table properties are checked up
+to 4 deep. Specify options.maxDepth to increase the depth.
 ]]
 DisplayUtil.assertPropTypes = function(props, propTypes, options)
+	local DevFlags = require('Module:DevFlags')
+	if not (DevFlags.matchGroupDev or DevFlags.forceTypeCheck) then

```

Comment:
```suggestion
	if not DevFlags.isDev() then

```

Seems like it would be much clearer

Reply:
I plan to rework `DevFlags` at some point in the future

Reply:
Please do it now

Reply:
I'm going to wait until after this round of changes to reduce merge conflicts.

Reply:
Alright, sounds good